apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+     * @notice Payable function to receive ETH
+     */
+    function addFunds() public payable {
+        // convert ETH to EtherToken
+        etherToken.wrapAndCall.value(msg.value)(address(finance), "Adding Funds");
+        LogFund(
+            msg.sender,
+            address(etherToken),
+            msg.value,
+            etherToken.balanceOf(this),
+            ""
+        );
+    }
+
+    /**
+     * @dev Allows make a simple payment from this contract to Finance,
+            to avoid locked tokens in contract forever.
+            This contract should never receive tokens with a simple transfer call,
+            but in case it happens, this function allows to recover them.
+     * @notice Allows to send tokens from this contract to Finance, to avoid locked tokens in contract forever
+     */
+    function depositToFinance(address token) public auth(OWNER_WRITE_ROLE) {
+        ERC20 tokenContract = ERC20(token);
+        uint256 value = tokenContract.balanceOf(this);
+        require(value >= 0);
+
+        // make an aprrovement for the same value to Finance
+        tokenContract.approve(address(finance), value);
+        // finally deposit those tokens to Finance
+        finance.deposit(tokenContract, value, "Adding Funds");
+    }
+
+    /**
+     * @dev Implement escape hatch mechanism. Avoids locked in contract forever
+     * @notice Implement escape hatch mechanism. Avoids locked in contract forever
+     */
+    function escapeHatch() public auth(OWNER_WRITE_ROLE) {
+        for (uint i = 0; i < globalTokensArray.length; i++) {
+            var token = globalTokensArray[i];
+            ERC20 tokenContract = ERC20(token);
+            uint256 tokenBalance = tokenContract.balanceOf(this);
+            // Avoid zero balances
+            if (tokenBalance == 0)
+                continue;
+            // send balance to owner
+            if (token != address(etherToken)) {
+                tokenContract.transfer(employer, tokenBalance);
+            } else {
+                etherToken.withdraw(employer, tokenBalance); // withdraw ether to receiver
+            }
+            LogFund(
+                employer,
+                token,
+                tokenBalance,
+                tokenContract.balanceOf(this),
+                ""
+            );
+        }
+        selfdestruct(employer); // send funds to organizer
+    }
+
+    /**
+     * @dev Implement escape hatch mechanism for critical case. Avoids locked in contract forever
+     * @notice Implement escape hatch mechanism for critical case.
+               Avoids locked in contract forever. If above function fails
+               (imagine, e.g., a bad token with a failing implementation of transfer),
+               this one would at least recover ETH (which btw should never remain in this contract).
+     */
+    function escapeHatch2() public auth(OWNER_WRITE_ROLE) {
+        selfdestruct(employer); // send funds to organizer
```

Comment:
Better than self destruct, just call `addFunds()` here, to ensure ETH is wrapped before being sent

Reply:
Good idea. Actually, I think we can even remove this function and just rely on `addFunds`.